core/src/test/java/com/amplifyframework/logging/AndroidLoggingPluginTest.java

```diff
@@ -131,6 +137,39 @@ public void noContentLoggedAtThresholdNone() {
         assertTrue(systemLog.getLines().isEmpty());
     }
 
+    /**
+     * When the application is not in a debuggable build, logs are not stored.
+     */
+    @Test
+    public void noLogsStored() {
+        Context mockContext = mock(Context.class);
+        ApplicationInfo mockAppInfo = mock(ApplicationInfo.class);
+        mockAppInfo.flags = 0;
+        when(mockContext.getApplicationInfo()).thenReturn(mockAppInfo);
+        AndroidLoggingPlugin plugin = new AndroidLoggingPlugin();
+        plugin.configure(new JSONObject(), mockContext);
+        Logger logger = plugin.forNamespace("logging-test");
+        logger.info("Info log");
+        assertNull(plugin.getLogs());
+    }
+
+    /**
+     * When the application is in a debuggable build, logs are stored.
+     */
+    @Test
+    public void logsAreStored() {
+        Context mockContext = mock(Context.class);
+        ApplicationInfo mockAppInfo = mock(ApplicationInfo.class);
+        mockAppInfo.flags = ApplicationInfo.FLAG_DEBUGGABLE;
+        when(mockContext.getApplicationInfo()).thenReturn(mockAppInfo);
```

Comment:
Elsewhere, I suggest making a utility to tell whether or not the app is debuggable. When I wrote it, I had it as a static utility class. However, it might simplify your test code if it is just package-local, open-for-extension, regular class:
```java
class AmplifyApp {
    private final WeakReference<Context> context;

    AmplifyApp(WeakReference<Context> context) {
        this.context = context;
    }

    boolean isDebuggable() {
        Context currentContext = context.get();
        return currentContext.blahblah();
    }
}
```
Or its test form:
```java
AmplifyApp amplifyApp = mock(AmplifyApp.class);
when(amplifyApp.isDebuggable()).thenReturn(true);
```

Reply:
After changing the code to add a BroadcastLogger and manage the loggers inside of LoggingCategory, I now only check if the app is in a debuggable build inside DeveloperMenuManager.enableDeveloperMenu, so I did not add a utility class for this.